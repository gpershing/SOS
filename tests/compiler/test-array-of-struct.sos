struct x = {int i}
struct y = {int i}

x distance1_x = {5}
x distance2_x = {6}
y distance1_y = {1}

array b = [distance1_x, {10}, distance1_y]
b[0] = distance2_x
b[1].i = 10

x new = b[0]
y new2 = b[1]



