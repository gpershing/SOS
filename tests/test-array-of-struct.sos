struct x = {i: int}
struct y = {i: int}

distance1_x: x = {5}
distance2_x: x = {6}
distance1_y: y = {1}

array x b = [distance1_x, {10}, distance1_y]
//array x b = [distanc1_x, {10}, {1}]
print(b[0].i) //5
print(b[1].i) //10
print(b[2].i) //1

b[0] = distance2_x
print(b[0].i) //6

b[0] = distance1_y
print(b[0].i) //1

b[0].i = 2
print(b[0].i) //2

