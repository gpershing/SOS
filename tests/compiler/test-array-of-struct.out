5
10
1
6
1
2
